src/codeu/chat/client/core/Controller.java

```diff
@@ -263,4 +263,52 @@ public Uuid changeAccessControl(Uuid userID, Uuid convoID, String username, User
 		}
 		return null;
 	}
+
+	// send an add bot request along with the id of the convo and the name of the bot that should be added to the convo
+	// it expects a boolean indicating whether or not a bot with that name was successfully added to the convo
+	// true = bot added, false = bot not added
+	@Override
+	public boolean addBot(Uuid convoId, String botName) {
```

Comment:
oooo we could return nil from the server if same bot is added twice. Will talk with @Priya411  about that.